### ISSUE TITLE
Should we update the condition for already_run?

### ISSUE THREAD

**Issue by Kelvin945** [jjjchens235/bing-rewards#241]
Currently the condition for already run is converting current and previous run time to "US/Alaska" timezone. If the time duration < 24 hours then return True. 

```
last_ran = self.__run_log.user_entries[-1].split(": ")[0]
last_ran_pst = datetime.strptime(last_ran, self.__DATETIME_FORMAT).replace(tzinfo=self.__LOCAL_TIMEZONE).astimezone(self.__PST_TIMEZONE)
run_datetime_pst = self.__run_datetime.astimezone(
    self.__PST_TIMEZONE
)
delta_days = (run_datetime_pst.date() - last_ran_pst.date()).days
is_already_ran_today = (
    (delta_days == 0 and last_ran_pst.hour >= self.__RESET_HOUR) or
    (delta_days == 1 and run_datetime_pst.hour < self.__RESET_HOUR)
)
```
However, for users using different timezone. One possible scenario will be: 
1. Run bot at night on day 1
2. Reward reset according to local timezone
3. Run bot at morning on day 2

In above scenario, the bot will not start since "US/Alaska" timezone might not pass reset hour yet. 
The possible fix I can think about will be:  
1. Using local timezone for comparing
2. Converting local timezone reset hour to "US/Alaska" timezone

Is this condition intended behavior or did I missing some parameters?

**Comment by Killerherts** [comment]
It seems the best fix would be to fetch everyone's local times. Do resets happen at same time for all timezones?

**Comment by Kelvin945** [comment]
> It seems the best fix would be to fetch everyone's local times. Do resets happen at same time for all timezones?

It seems they reset at different time.According to my observation, daily set reset after 12pm localtime. 

Not so sure about the rules for earnings and more activities. In my case they reset after next day 2 am. 

Anyone have similar experience?

**Comment by Lohengrim** [comment]
Yes, i've notice the same thing. Didn't realize that it could be caused by the timezone. 
As a workaround I was deleting logs folder daily just before my next scheduled run.

**Comment by JourneyOver** [comment]
I'm in Central timezone

daily set reset for me at midnight (daily set goes by persons timezone and always resets at midnight local time)

Searches and More activities reset at a set time. From what I can tell and searching around the times are as: (times below are in Daylight saving time)
```
4AM EDT
3AM CDT
2AM MDT
1AM PDT
```
though it seems the More activities bit sometimes takes a bit longer to reset than the searches for some reason but it's seems to be rare.

**Comment by AnthonyBe** [comment]
Being located in Australia, this facet of the operation of the script has always been confusing! 

It would be great if it could register the local TZ of the user and then vector that into feedback on whether searches or activities have reset.
